### PR TITLE
expressionUpdates do not contain Method Pointer of operators

### LINKED ISSUE #7520: expressionUpdates do not contain Method Pointer of operators.
I have a project, where next to the starting "Press tab to..." node, I added `3 +` node.

Then I inspected the network tab and look through "expressionUpdate" messages IDE had received from the engine. No expression contained a method pointer for "+" operator (there was no "+" character in all messages).

`3 + _` or even `3 + 4` also had no method pointer. Tested on 2023.2.1-nightly.2023.8.4 engine.

This task is blocking #7107. The IDE implementation is on branch `wip/farmaazon/placeholders-for-operators` - you can test if your fix makes the argument names appear in nodes like `3 +`.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #7520

Changelog:
- update: SectionsToBinOp compiler pass produces function application for left sections
- refactor: simplify the registration of builtin methods

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

**Comment by 4e6** [comment]
#6959 should fix the other part of the issue. Waiting for it to be closed

**Review by hubertp — APPROVED** [review]

**Review by jdunkerley — APPROVED** [review]

File: engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java [hunk 1]
```diff
@@ -29,13 +31,9 @@
 import org.enso.interpreter.runtime.error.PanicException;
 import org.enso.interpreter.runtime.error.PanicSentinel;
 import org.enso.interpreter.runtime.state.State;
+import org.enso.interpreter.runtime.tag.AvoidIdInstrumentationTag;
 import org.enso.interpreter.runtime.tag.IdentifiedTag;
 import org.enso.interpreter.runtime.type.Constants;
-import org.enso.interpreter.runtime.Module;
-
-import java.util.function.Consumer;
```
Comment (JaroslavTulach) [inline engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java:36]: Heh, someone's IDE is reordering imports after changing a comment...

File: engine/runtime/src/test/java/org/enso/compiler/CompilerTest.java [hunk 2]
```diff
@@ -10,8 +13,6 @@
 import org.enso.compiler.core.IR;
 import org.enso.compiler.core.ir.Module;
 import org.junit.AfterClass;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
```
Comment (JaroslavTulach) [inline engine/runtime/src/test/java/org/enso/compiler/CompilerTest.java:14]: Just an update of imports without any other change isn't really justified.

File: engine/runtime/src/main/scala/org/enso/compiler/pass/desugar/SectionsToBinOp.scala [hunk 3]
```diff
@@ -146,21 +145,16 @@ case object SectionsToBinOp extends IRPass {
...
             function             = op,
-            arguments            = List(arg, rightCallArg),
+            arguments            = List(arg),
             hasDefaultsSuspended = false,
-            location             = None,
+            location             = loc,
             passData,
             diagnostics
           )
-
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/scala/org/enso/compiler/pass/desugar/SectionsToBinOp.scala:157]: This is the fix?

>  SectionsToBinOp compiler pass produces function application for left sections

What source code this change affects? The expression...

> `(1 +)`

...will now become `Application.Prefix(function="+", arguments=[1])`. OK, that seems correct.

**Review by JaroslavTulach — APPROVED** [review]
The change in `SectionsToBinOp` seems correct to me.

**Pull request merged**